File: surface/widgets/compose/components/widgets/navigation/Breadcrumb.kt
// ============================================================
// Clef Surface Compose Widget — Breadcrumb
//
// Hierarchical location trail for Jetpack Compose.
// Renders a horizontal path like "Home > Products > Item"
// with dimmed non-current items and bold for the current page.
// Maps breadcrumb.widget anatomy (root, list, item, link,
// separator, currentPage) to Row with Text composables.
// See Architecture doc Section 16.
// ============================================================

package clef.surface.compose.components.widgets.navigation

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

// --------------- Types ---------------

data class BreadcrumbItem(
    val label: String,
    val href: String? = null,
    val current: Boolean? = null,
)

// --------------- Component ---------------

/**
 * Hierarchical location trail.
 *
 * @param items Ordered breadcrumb trail items.
 * @param separator Separator character between items.
 * @param onItemClick Callback when a non-current item is clicked.
 * @param modifier Modifier for the root layout.
 */
@Composable
fun Breadcrumb(
    items: List<BreadcrumbItem>,
    separator: String = "\u203A",
    onItemClick: ((BreadcrumbItem) -> Unit)? = null,
    modifier: Modifier = Modifier,
) {
    if (items.isEmpty()) return

    Row(
        modifier = modifier,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        items.forEachIndexed { index, item ->
            val isLast = index == items.lastIndex
            val isCurrent = item.current ?: isLast

            if (isCurrent) {
                Text(
                    text = item.label,
                    style = MaterialTheme.typography.bodyMedium,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSurface,
                )
            } else {
                Text(
                    text = item.label,
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.clickable { onItemClick?.invoke(item) },
                )
            }

            if (!isLast) {
                Text(
                    text = " $separator ",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(horizontal = 4.dp),
                )
            }
        }
    }
}

File: surface/widgets/swiftui/components/widgets/navigation/Breadcrumb.swift
// ============================================================
// Clef Surface SwiftUI Widget — Breadcrumb
//
// Hierarchical location trail rendered as a horizontal path
// with separator characters between items.
// ============================================================

import SwiftUI

// --------------- Types ---------------

struct BreadcrumbItem: Identifiable {
    let id = UUID()
    let label: String
    var href: String? = nil
    var current: Bool? = nil
}

// --------------- Component ---------------

/// Breadcrumb view rendering a hierarchical location trail.
///
/// - Parameters:
///   - items: Ordered breadcrumb trail items.
///   - separator: Separator character between items.
///   - onItemClick: Callback when a non-current item is clicked.
struct BreadcrumbView: View {
    var items: [BreadcrumbItem]
    var separator: String = "\u{203A}"
    var onItemClick: ((BreadcrumbItem) -> Void)? = nil

    var body: some View {
        HStack(spacing: 0) {
            ForEach(Array(items.enumerated()), id: \.element.id) { index, item in
                let isLast = index == items.count - 1
                let isCurrent = item.current ?? isLast

                if isCurrent {
                    Text(item.label)
                        .font(.body)
                        .fontWeight(.bold)
                        .foregroundColor(.primary)
                } else {
                    SwiftUI.Button(action: {
                        onItemClick?(item)
                    }) {
                        Text(item.label)
                            .font(.body)
                            .foregroundColor(.secondary)
                    }
                    .buttonStyle(.plain)
                }

                if !isLast {
                    Text(" \(separator) ")
                        .font(.body)
                        .foregroundColor(.secondary)
                        .padding(.horizontal, 4)
                }
            }
        }
        .accessibilityElement(children: .contain)
    }
}
